### PR TITLE
[MM-37693]: added test for deleting a root post

### PR DESCRIPTION
**Pull request by michelengelen**
#### Summary
added a test case for deleting a root post and checking the global threads tab. 
There is some "wonky" behavior when deleting a root post as described below.
Since we do not really have a spec for this I outline what I would expect with our current approach:

---

### in main channel view
**Expected**
it shows `(message deleted)` for the root post and no further info on the Thread and participants

**Actual**
it shows `(message deleted)` for the root post and the ThreadFooter still shows all elements (including reply count and participating users)

---

### in RHS
**Expected** (options)
1. no view (or at least one like for archived channels: "posting is not possible ...")
2. opening a ThreadView should not be possible/opened ThreadView should close

**Actual**
opening a ThreadView is possible and the input field is active (although posting is not possible)

---

### in global threads overview
**Expected** (options)
thread should not show up

**Actual**
when the root post is deleted while the global thread view is opened the thread item shows all infos of a regular thread (replies count, participants) but with a `(message deleted)` post as root, but when refreshing the thread is being removed from the view. This is the case I implemented in this test as well, but should be adjusted once we have a more solidified solution for it!

#### Ticket Link
[MM-37693](https://mattermost.atlassian.net/browse/MM-37693)

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/tests/support/api_commands.js
Comment (michelengelen) [reply e2e/cypress/tests/support/api_commands.js:91]: I can build that in ... no problem

**Comment by saturninoabril** [comment]
I'll update the branch then see if E2E slash command will work fine.

File: e2e/cypress/tests/support/api_commands.js
Comment (michelengelen) [reply e2e/cypress/tests/support/api_commands.js:91]: > I can build that in ... no problem

@saturninoabril As I branched off of this for MM-T4446 (in #10060) I fixed it in there, but will port the fix over to this as well.

File: e2e/cypress/tests/integration/collapsed_reply_threads/root_post_spec.js
Comment (saturninoabril) [inline e2e/cypress/tests/integration/collapsed_reply_threads/root_post_spec.js:55]: Missed to mention but I suggests to match the title with the Zephyr. SO either change here or update the title in Zephyr.
```suggestion
    it('MM-T4445 CRT - Delete root post', () => {
```

**Review by saturninoabril — APPROVED** [review]
Thanks @michelengelen! All good except for last comment. Approving now.

E2E at https://automation-dashboard.vercel.app/cycles/de2463bf-abf3-438b-b9b2-32155621e4f6

**Comment by michelengelen** [comment]
switched to a new feature branch for collecting all new tests before merging them into master. The collection ticket is [MM-37693](https://mattermost.atlassian.net/browse/MM-37693)

**Comment by metanerd** [comment]
Whew, seems like it was rate limited the first 2 times. We need to fix this rate limit issue. It is annoying.

**Comment by metanerd** [comment]
Ah, feature branch into feature branch setup is not supported when running e2e tests. FYI

**Pull request merged**
